app/assets/javascripts/school_administrator_dashboard/dashboard_components/discipline_dashboard/SchoolWideDisciplineIncidents.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import SchoolDisciplineDashboard from './SchoolDisciplineDashboard';
+import DashboardHelpers from '../DashboardHelpers';
+import {latestNoteDateText} from '../../../helpers/latestNoteDateText';
+
+class SchoolWideDisciplineIncidents extends React.Component {
+
+  schoolDisciplineEvents() {
+    //structures the discipline incident data for easy grouping
+    const students = this.props.dashboardStudents;
+    const startDate = DashboardHelpers.schoolYearStart();
+    let schoolDisciplineEvents = [];
+    students.forEach((student) => {
+      student.discipline_incidents.forEach((incident) => {
+        if (moment(incident.occurred_at).isSameOrAfter(moment(startDate))) {
+          schoolDisciplineEvents.push({
+            student_id: incident.student_id,
+            location: incident.incident_location,
+            time: incident.has_exact_time ? moment(incident.occurred_at).format("h A") : "Not Logged",
+            classroom: student.homeroom_label,
+            student_grade: student.grade,
+            day: moment(incident.occurred_at).format("ddd"),
+            offense: incident.incident_code,
+            student_race: student.race,
+            occurred_at: incident.occurred_at,
+            last_sst_date_text: latestNoteDateText(300, student.event_notes)
+          });
```

Comment:
Can you say more about why you're mapping these items into a different shape here?  I was expecting this to happen in the rendering and I'm wondering if this is maybe because of how you've factored the charts.

Reply:
@kevinrobinson It's not necessary to do this here. This component could be combined with the disciplineDashboard component, but it made more sense to me to just get the items once and pass them as a prop to the component that needs to manipulate them. Is that what you meant by having this happen in the render?

Reply:
Ah yeah I see what you're saying.  This component is transforming the data from the server, and the `SchoolDisciplineDashboard` component is visualizing it in different ways.  So you're putting that transformation here, and then the other component doesn't have to transform further, and can render that data in a few different ways, is that right?

The main thing I'm commenting on here is that those two pieces of code are tightly coupled but not located next each other - the transformation here is only correct if it's doing what the UI code in `SchoolDisciplineDashboard` expects, and the UI code in `SchoolDisciplineDashboard` is only correct if it's working with data in the shape that this component is making.  This might be fine here, but if you keep the distance between them, maybe add in the `propTypes` about what particular fields need to be passed to `SchoolDisciplineDashboard`.  Then that adds a small check that the "seam" between those two components works correctly.

Reply:
@kevinrobinson yeah, that's it. AFAIK it won't be such a big deal given the likely number of discipline incidents, but separating out this transform made a big difference in the absence dashboard's performance. 

Good point about making the connection clear. I'll make that more explicit.

Reply:
Gotcha.  Yeah, caching repeated computations is a great way to make this faster!  The comment here suggests that the code is doing this for easy grouping, so it might be good to edit or add a one-line comment explaining how it impacts performance.